### PR TITLE
x11: Expose the text/global scaling factor as the display content scale

### PR DESCRIPTION
**Pull request by Kontrabant**
Expose the text scaling factor (aka the global scale factor on KDE) as the display content scale value so applications can scale themselves as necessary.

If D-Bus is unavailable or retrieving the setting fails, fall back to trying the legacy GDK_SCALE envvar before reverting to the default 1.0 value.

### CODE REVIEW

File: src/video/x11/SDL_x11modes.c [hunk 1]
```diff
@@ -39,6 +39,91 @@
...
+        {
+            const char *scale_str = SDL_getenv("GDK_SCALE");
+            if (scale_str) {
+                scale_factor = SDL_atoi(scale_str);
+            }
+        }
+
+        /* Nothing or a bad value, just fall back to 1.0 */
+        if (scale_factor <= 0.0) {
+            scale_factor = 1.0;
```
Comment (slouken) [inline src/video/x11/SDL_x11modes.c:120]: 0.0f and 1.0f?

Comment (Kontrabant) [reply src/video/x11/SDL_x11modes.c:120]: `scale_factor` is a double as the gsettings value is stored as a double and needs a double type for retrieval. It's truncated to a float on return.

**Comment by slouken** [comment]
Yes, please merge this when you're ready. This will be very helpful to people using the content scale on Linux. ;)

**Comment by Kontrabant** [comment]
No problem, just wanted to wait to make sure it didn't collide with anything you were working on.

**Comment by slouken** [comment]
Great. Is there a notification when the scale changes?

**Comment by Kontrabant** [comment]
Changing the scale on X11 generally requires a logout→login cycle, so changing it on the fly isn't really a concern. When I tested it on KDE, the changed scale value wasn't reflected at all until doing so.

**Comment by lostgoat** [comment]
I think there might be some value in subscribing to change events.

As @Kontrabant mentioned, usually an app restart is required to pickup the scale change. For the KDE desktop itself this involves a logout->login cycle, but other client apps will pickup the scale change when closed/opened.  E.g.:

1. Set KDE scaling slider to 100%
2. Open the dolphin file manager
3. Set KDE scaling to 200% + Apply
4. Close dolphin
5. Start dolphin (should now show in 200% scale)

It looks like the portal interface supports a `SettingChanged` signal:
https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.Settings

I think it would be great to subscribe to that event and generate the corresponding SDL event to inform of a scale change.

**Comment by Kontrabant** [comment]
> It looks like the portal interface supports a `SettingChanged` signal:
> https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.Settings

Ah, I was under the impression that it only worked for the mentioned theme color setting, but it turns out to work for the scale factor as well :).  Pull incoming shortly.

**Pull request merged**